predicators/ground_truth_models/cover/nsrts.py

```diff
@@ -461,9 +483,17 @@ def place_on_target_sampler(state: State, goal: Set[GroundAtom],
         def place_on_bumpy_sampler(state: State, goal: Set[GroundAtom],
                                    rng: np.random.Generator,
                                    objs: Sequence[Object]) -> Array:
-            del state, objs, goal  # unused
-            lb = CFG.bumpy_cover_bumpy_region_start
-            ub = 1.0
+            del goal  # unused
+            b, = objs
+            w = state.get(b, "width") / 2
+            lb = CFG.bumpy_cover_bumpy_region_start + w
+            ub = 1.0 - w
+            # TODO: rejection sample here, checking for possible collisions
```

Comment:
@NishanthJKumar if you wanna tackle this, that would be helpful!

Reply:
yup - i got it!